src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -194,7 +189,7 @@ impl<'top> LazyRawValue<'top, BinaryEncoding_1_1> for &'top LazyRawBinaryValue_1
             return Ok(ValueRef::Int(int));
         }
         if self.is_null() {
-            return Ok(ValueRef::Null(self.ion_type()));
+            return self.read_null()?.resolve(context);
```

Comment:
The `resolve` method requires a giant `match` over all of the Ion types (see below). Since we already know it's a (possibly typed) `null`, I suggest `return ValueRef::Null(self.read_null())?;`

![image](https://github.com/user-attachments/assets/3a28e088-9236-40be-b59f-0ccc41dfd4a8)